### PR TITLE
Js lesson 4

### PR DESCRIPTION
**Pull request by IsakovaL**

### CODE REVIEW

File: hw/js-lesson-3/scripts.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+let shift = +prompt("enter shift"); //3
+let movedPart = numberString.substring(0, shift);//123
+
+numberString += movedPart;//"123456789123"
+numberString = numberString.substring (shift, numberString.length);
+alert(numberString); //456789123
+
+// 8 Зациклить вывод дней недели таким образом: «День недели. Хотите увидеть следующий день?» и так до тех пор, пока пользователь нажимает OK.
+
+while (true) {
```
Comment (idovbush) [inline hw/js-lesson-3/scripts.js:139]: your cycles is looped here

Comment (idovbush) [inline hw/js-lesson-3/scripts.js:139]: but where you show to user day of the week? you only show him question

File: hw/js-lesson-4/scripts.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// 1 Написать функцию, которая принимает 2 числа и возвращает -1, если первое меньше, чем второе; 1 – если первое больше, чем второе; и 0 – если числа равны.
+
+function conpare(a, b) {
```
Comment (idovbush) [inline hw/js-lesson-4/scripts.js:3]: compare

File: hw/js-lesson-4/scripts.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    return (n != 1) ? n * factorial(n - 1) : 1;
+  }
+factorial(3)
+
+// 3 Написать функцию, которая принимает три отдельные цифры и превращает их в одно число. Например: цифры 1, 4, 9 превратятся в число 149.
+
+function oneNumber(a, b, c) {
+
+  return  '' + a + b + c;
```
Comment (idovbush) [inline hw/js-lesson-4/scripts.js:30]: why don't you use something like `${a+b+c}` but even in this case it's not a number? it's a string

File: hw/js-lesson-4/scripts.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+       console.log("It is a perfect number.");
+        } 
+     else
+        {
+       console.log("It is not a perfect number.");
+        }   
+ } 
+isPerfect(6); 
+
+// 6 Написать функцию, которая принимает минимальное и максимальное значения для диапазона, и выводит только те числа из диапазона, которые являются совершенными.
```
Comment (idovbush) [inline hw/js-lesson-4/scripts.js:69]: why don't you finish this task?

File: hw/js-lesson-3/scripts.js (hunk 1 above)
Comment (IsakovaL) [reply hw/js-lesson-3/scripts.js:139]: when confirm it shows the day of the week and proposes user to show another day

File: hw/js-lesson-4/scripts.js (hunk 3 above)
Comment (IsakovaL) [reply hw/js-lesson-4/scripts.js:30]: to join all numbers in one I had no other idea but to make a string from them

File: hw/js-lesson-4/scripts.js (hunk 4 above)
Comment (IsakovaL) [reply hw/js-lesson-4/scripts.js:69]: broke my brain when reached this task

File: hw/js-lesson-4/scripts.js (hunk 3 above)
Comment (idovbush) [reply hw/js-lesson-4/scripts.js:30]: превращает их в одно число. Look accurately to description of the task

**Pull request closed without merging**